l1-contracts/src/core/messagebridge/Inbox.sol

```diff
@@ -0,0 +1,160 @@
+   * @dev Will emit `MessageAdded` with data for easy access by the sequencer
+   * @dev msg.value - The fee provided to sequencer for including the entry
+   * @param _recipient - The recipient of the entry
+   * @param _deadline - The deadline to consume a message. Only after it, can a message be cancalled.
+   * @param _content - The content of the entry (application specific)
+   * @param _secretHash - The secret hash of the entry (make it possible to hide when a specific entry is consumed on L2)
+   * @return The key of the entry in the set
+   */
+  function sendL2Message(
+    L2Actor memory _recipient,
+    uint32 _deadline,
+    bytes32 _content,
+    bytes32 _secretHash
+  ) external payable returns (bytes32) {
+    uint64 fee = uint64(msg.value);
+    L1ToL2Msg memory message = L1ToL2Msg({
+      sender: L1Actor(msg.sender, block.chainid),
+      recipient: _recipient,
+      content: _content,
+      secretHash: _secretHash,
+      deadline: _deadline,
+      fee: fee
+    });
+
+    bytes32 key = computeMessageKey(message);
+    _insert(key, fee, _deadline);
+
+    emit MessageAdded(
+      key,
+      message.sender.actor,
+      message.recipient.actor,
+      message.sender.chainId,
+      message.recipient.version,
+      message.deadline,
+      message.fee,
+      message.content
+    );
+
+    return key;
+  }
+
+  /**
+   * @notice Cancel a pending L2 message
+   * @dev Will revert if the deadline have not been crossed - message only cancellable past the deadline
+   *  so it cannot be yanked away while the sequencer is building a block including it
+   * @dev Must be called by portal that inserted the entry
+   * @param _message - The content of the entry (application specific)
+   * @param _feeCollector - The address to receive the "fee"
+   * @return entryKey - The key of the entry removed
+   */
+  function cancelL2Message(L1ToL2Msg memory _message, address _feeCollector)
+    external
+    returns (bytes32 entryKey)
+  {
+    if (msg.sender != _message.sender.actor) revert Inbox__Unauthorized();
+    if (_message.deadline <= block.timestamp) revert Inbox__NotPastDeadline();
+    entryKey = computeMessageKey(_message);
+    _consume(entryKey);
+    feesAccrued[_feeCollector] += _message.fee;
+    emit L1ToL2MessageCancelled(entryKey);
+  }
+
+  /**
+   * @notice Batch consumes entries from the Inbox
+   * @dev Only callable by the rollup contract
+   * @dev Will revert if the message is already past deadline
+   * @param entryKeys - Array of entry keys (hash of the messages)
+   * @param _feeCollector - The address to receive the "fee"
+   */
+  function batchConsume(bytes32[] memory entryKeys, address _feeCollector) external onlyRollup {
+    uint256 totalFee = 0;
+    for (uint256 i = 0; i < entryKeys.length; i++) {
+      // TODO: Combine these to optimise for gas.
+      Entry memory entry = get(entryKeys[i]);
+      if (entry.deadline > block.timestamp) revert Inbox__PastDeadline();
```

Comment:
No! If you are skipping and emitting an event this is handled offchain, but it would still be updating the rollup state. With rollup state now being updated with unvalidated msgs, the sequencer is essentially granted god mode and can impersonate any portal contract -> steal all funds.